Source/WebCore/css/CSSSelector.cpp

```diff
@@ -337,32 +338,68 @@ std::optional<CSSSelector::PseudoElement> CSSSelector::parsePseudoElementName(St
 }
 
 // FIXME: We should eventually deduplicate this with CSSSelectorParser::consumePseudo() somehow.
-std::optional<PseudoId> CSSSelector::parsePseudoElement(const String& input, const CSSSelectorParserContext& context)
+std::pair<bool, std::optional<Style::PseudoElementIdentifier>> CSSSelector::parsePseudoElement(const String& input, const CSSSelectorParserContext& context)
```

Comment:
This would be better as a static function in CSSSelectorParser (does not need to be in this patch).

Also as the comment says the code should be shared with CSSSelectorParser::consumePseudo.